Resources/public/js/views/universaldiscovery/ez-universaldiscoveryselectedview.js

```diff
@@ -22,16 +27,50 @@ YUI.add('ez-universaldiscoveryselectedview', function (Y) {
      * @constructor
      * @extends eZ.TemplateBasedView
      */
-    Y.eZ.UniversalDiscoverySelectedView = Y.Base.create('universalDiscoverySelectedView', Y.eZ.TemplateBasedView, [], {
+    Y.eZ.UniversalDiscoverySelectedView = Y.Base.create('universalDiscoverySelectedView', Y.eZ.TemplateBasedView, [Y.eZ.AsynchronousView], {
         events: {
             '.ez-ud-selected-confirm': {
                 'tap': '_confirmSelected',
             }
         },
 
         initializer: function () {
+            var imageField;
+            this._fireMethod = function () {};
+            this._triggerAttribute = 'imageField';
+
+            this.after('imageStateChange', this._uiSetState);
+            this.after('viewStateChange', this._uiSetState);
+            this.after('loadImageVariation', function () {
+                this._set('imageState', STATE_IMAGE_LOADING);
+            });
+
+            this.after('imageVariationChange', function (e) {
+                if (this.get('imageVariation')) {
+                    this._set('imageState', STATE_IMAGE_LOADED);
+                    this._renderImageVariation();
+                }
+            });
+
+            this._errorHandlingMethod = function () {
+                this._set('imageState', STATE_IMAGE_ERROR);
+            };
+            
             this.after('contentStructChange', function (e) {
                 this._setConfirmButtonState(e.newVal);
+                if ( this.get('contentStruct') && this.get('contentStruct').content ) {
```

Comment:
you should separate the case for the translations from the image handling case. Also this would deserve to be extracted in different methods.